modules/consensus/synchronize.go

```diff
@@ -91,8 +107,13 @@ func (cs *ConsensusSet) threadedReceiveBlocks(conn modules.PeerConn) error {
 
 		// Integrate the blocks into the consensus set.
 		for _, block := range newBlocks {
-			acceptErr := cs.AcceptBlock(block)
-
+			// Call managedAcceptBlock instead of AcceptBlock so as not to broadcast
+			// every block.
+			acceptErr := cs.managedAcceptBlock(block)
+			// Only broadcast the last block if a block was accepted.
```

Comment:
this comment is a little confusing, because it makes it seem like you are doing the broadcasting immediately, instead of setting a flag that will trigger a broadcast in the defer statement.